healthcare/api-client/v1/datasets/datasets_test.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2018 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import time
+
+import datasets
+
+cloud_region = 'us-central1'
+project_id = os.environ['GOOGLE_CLOUD_PROJECT']
+
+dataset_id = 'test-dataset-{}'.format(int(time.time()))
+destination_dataset_id = 'test-destination-dataset-{}'.format(int(time.time()))
+keeplist_tags = 'PatientID'
+time_zone = 'UTC'
+
+
+def test_CRUD_dataset(capsys):
+    datasets.create_dataset(
+        project_id,
+        cloud_region,
+        dataset_id)
+
+    datasets.get_dataset(
+        project_id, cloud_region, dataset_id)
+
+    datasets.list_datasets(
+        project_id, cloud_region)
+
+    # Test and also clean up
+    datasets.delete_dataset(
```

Comment:
Having cleanup in the test itself is not reliable.

Can you create a fixture which yield the name, and make sure it will be deleted after the test?

Reply:
Created test_dataset fixture.

Reply:
I think you need one more fixture, something like:

```
@pytest.fixture(scope="module")
def crud_dataset_id():
    yield dataset_id
    # Clean up
    @retry(#params for retryihg N times only for certain errors e.g. not retrying for not found)
    def clean_up():
        datasets.delete_dataset(project_id, cloud_region, dataset_id)
```

Then use this in CRUD test.

We are seeing lots of test failures in fixtures too, so I'd prefer having retries in fixture for more reliable tests.

Reply:
I replied to this but for some reason it doesn't look like my comment was saved. I created crud_dataset_id and dest_dataset_id fixtures that try to delete the original dataset and the destination dataset if exception is anything other than 404. I chose 404 because I can't think of another error we would choose to not retry on.

Reply:
Looking great, thanks